src/main/java/ui/BoardAutoCreator.java

```diff
@@ -74,6 +74,70 @@ private void createMilestoneBoard() {
                         "It is saved under the name \"" + boardName + "\".");
     }
 
+    private void createWorkAllocationBoard() {
+        logger.info("Creating " + WORK_ALLOCATION + " board");
+
+        List<TurboUser> userList = ui.logic.getRepo(ui.logic.getDefaultRepo()).getUsers();
+
+        if (userList.isEmpty()) {
+            DialogMessage.showInformationDialog("Auto-create Board - " + WORK_ALLOCATION,
+                    WORK_ALLOCATION + " board cannot be created as your" +
+                            " account has no push access to the default repo.");
+
+            return;
+        }
+
+        int noOfPanelsToBeGenerated = Math.min(MAX_WORK_ALLOCATION_PANELS, userList.size());
+
+        List<PanelInfo> panelData = generatePanelInfoFromTurboUsers(userList.subList(0, noOfPanelsToBeGenerated),
+                "Work allocated to %s", "assignee:%s sort:milestone,status");
+
+        String boardName = Utility.getNameClosestToDesiredName(WORK_ALLOCATION, getBoardNames());
+
+        createBoard(panelData, boardName);
+
+        DialogMessage.showInformationDialog("Auto-create Board - " + WORK_ALLOCATION,
+                WORK_ALLOCATION + " board has been created and loaded.\n\n" +
+                        "It is saved under the name \"" + boardName + "\".");
+    }
+
+    private List<PanelInfo> generatePanelInfoFromTurboUsers(List<TurboUser> userList,
+                                                            String nameTemplate, String filterTemplate) {
+        List<PanelInfo> panelData = new ArrayList<>();
+        String panelName, panelFilter;
+
+        for (TurboUser eachUser : userList) {
```

Comment:
We could write this as a `map` over the collection of users. Also the parameter can probably be named `users`, so it isn't tied to its implementation.